onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc

```diff
@@ -0,0 +1,89 @@
+  if (bc.IsInput0Scalar()) {
+    while (output)
+      input0scalar(output.NextEigenOutput(), bc.NextScalar0(), bc.NextEigen1(), A_scale, B_scale, C_scale, A_zero_point, B_zero_point, C_zero_point);
+  } else if (bc.IsInput1Scalar()) {
+    while (output)
+      input1scalar(output.NextEigenOutput(), bc.NextEigen0(), bc.NextScalar1(), A_scale, B_scale, C_scale, A_zero_point, B_zero_point, C_zero_point);
+  } else {
+    while (output)
+      general(output.NextEigenOutput(), bc.NextEigen0(), bc.NextEigen1(), A_scale, B_scale, C_scale, A_zero_point, B_zero_point, C_zero_point);
+  }
+}
+
+template <typename T, typename Input0Scalar, typename Input1Scalar, typename General>
+Status QLinearBroadcastTwo(OpKernelContext& context, Input0Scalar input0scalar, Input1Scalar input1scalar, General general) {
+  const float A_scale = *(context.Input<Tensor>(1)->Data<float>());
+  const T A_zero_point = (nullptr == context.Input<Tensor>(2)) ? static_cast<T>(0) : *(context.Input<Tensor>(2)->template Data<T>());
+  const float B_scale = *(context.Input<Tensor>(4)->Data<float>());
+  const T B_zero_point = (nullptr == context.Input<Tensor>(5)) ? static_cast<T>(0) : *(context.Input<Tensor>(5)->template Data<T>());
+  const float C_scale = *(context.Input<Tensor>(6)->Data<float>());
+  const T C_zero_point = (nullptr == context.Input<Tensor>(7)) ? static_cast<T>(0) : *(context.Input<Tensor>(7)->template Data<T>());
+
+  TBroadcaster<T, T> bc(*context.Input<Tensor>(0), *context.Input<Tensor>(3));
+  TBroadcastOutput<T> output(bc.GetSpanSize(), *context.Output(0, bc.GetOutputShape()));
+  QLinearBroadcastLoop(bc, output, input0scalar, input1scalar, general, A_scale, B_scale, C_scale, static_cast<int>(A_zero_point), static_cast<int>(B_zero_point), static_cast<int>(C_zero_point));
+  return Status::OK();
+}
+
+template <typename T>
+Status QLinearAdd<T>::Compute(OpKernelContext* context) const {
+  return QLinearBroadcastTwo<T>(
+      *context,
+      [](EigenVectorMap<T> output, T input0, ConstEigenVectorMap<T> input1, float A_scale, float B_scale, float C_scale, int A_zero_point, int B_zero_point, int C_zero_point) {
+        float a_value = A_scale * (static_cast<int>(input0) - A_zero_point);
+        output = (((((input1.array().template cast<int>() - B_zero_point).template cast<float>() * B_scale) + a_value) / C_scale).round() - static_cast<float>(C_zero_point)).template cast<T>();
+      },
+      [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T input1, float A_scale, float B_scale, float C_scale, int A_zero_point, int B_zero_point, int C_zero_point) {
+        float b_value = B_scale * (static_cast<int>(input1) - B_zero_point);
+        output = (((((input0.array().template cast<int>() - A_zero_point).template cast<float>() * A_scale) + b_value) / C_scale).round() - static_cast<float>(C_zero_point)).template cast<T>();
```

Comment:
> static_cast<float> [](start = 139, length = 19)

round already cast the 1st operand to int. There is no need to cast to float here.

Reply:
Yes, could not cast here. Yet just for sure.

Reply:
so the cast is zero cost or not?